server.js

```diff
@@ -1338,64 +1487,112 @@ class Callbacks {
     }
 
     static cApplySettings(socket, key, settings) {
-        // special case: "dictionaryId" (it changes ranges for other settings)
+        // special case: "dictionaryId" (it changes ranges for other settings) (also "termCondition" should be processes out of turn (and "wordsetType"))
         let warnWordsDecrease = false;
         let warnTurnDefault = false;
         let warnWordsDefault = false;
-        if ("termCondition" in settings) {
-            if (typeof(rooms[key].settings["termCondition"]) !== typeof(settings["termCondition"])) {
+        let ignoreTermCondition = false;
+        if ("wordsetType" in settings) {
+            if (typeof(rooms[key].settings["wordsetType"]) !== typeof(settings["wordsetType"])) {
```

Comment:
Тут на значение проверять надо, а не на тип:
```suggestion
            if (!(rooms[key].settings["wordsetType"] in {"serverDictionary": null, "hostDictionary": null, "playerWords": null})) {
```

Reply:
Или как у `termCondition` двухступенчатую проверку сделать. Или ты на скорую руку?

Reply:
Да, сейчас поправлю